### PR TITLE
Update webpack and webpack-related deps

### PR DESCRIPTION
**Pull request by askoufis**
Originally just wanted to upgrade `webpack-dev-server` in the `webpack-react` example to address #1280, but that snowballed into upgrading some other deps too. Kept it scoped to just webpack-related stuff (mostly loaders) that were at least a major version behind.

Will followup afterwards with some more dep upgrades to the webpack example.

Followed [this migration guide](https://github.com/webpack/webpack-dev-server/blob/48f56d57775dc4cf78bfcf51c3761adf852b0310/migration-v4.md) for `webpack-dev-server` v3 -> v4.

### CODE REVIEW

File: packages/webpack-plugin/src/compiler.ts [hunk 1]
```diff
@@ -81,7 +81,7 @@ function compileVanillaSource(
     );
     const ExternalsPlugin = compat.getExternalsPlugin(loader._compiler);
 
-    new NodeTemplatePlugin(outputOptions).apply(childCompiler);
+    new NodeTemplatePlugin().apply(childCompiler);
```
Comment (askoufis) [inline packages/webpack-plugin/src/compiler.ts:84]: `outputOptions` contained just a `filename` property. Looking at both the webpack 4 and 5 docs, this was never a valid property for this plugin. It was causing a type error, so I removed it.

File: test-helpers/src/startFixture/webpack.ts [hunk 2]
```diff
@@ -96,25 +95,32 @@ export const startWebpackFixture = (
     });
     const compiler = webpack(config);
 
-    // @ts-expect-error - webpack version mismatch
-    const server = new WDS(compiler, {
-      hot,
-      stats: logLevel,
-      noInfo: true,
```
Comment (askoufis) [inline test-helpers/src/startFixture/webpack.ts:103]: These were removed in `webpack-dev-server` v4.

Comment (mrm007) [reply test-helpers/src/startFixture/webpack.ts:103]: Pushed [`86f7733` (#1287)](https://github.com/vanilla-extract-css/vanilla-extract/pull/1287/commits/86f773361151a15073e1dcc552580636b7d2462c) to restore the logging level. Feel free to drop the commit if you disagree.

**Review by mrm007 — APPROVED** [review]

File: test-helpers/src/startFixture/webpack.ts (hunk 2 above)
Comment (askoufis) [reply test-helpers/src/startFixture/webpack.ts:103]: LGTM. I tried with something similar but kept getting type errors. Honestly not too concerned with the fixture log output, but thanks 👍.

**Pull request merged**